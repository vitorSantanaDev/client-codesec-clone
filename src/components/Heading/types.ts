import { ReactNode } from 'react'

export type IViewProps = {
  as?: 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6'
  size?:
    | 'small'
    | 'medium'
    | 'large'
    | 'xlarge'
    | 'xxlarge'
    | 'huge'
    | 'xhuge'
    | 'xxhuge'
  color?: string
  children: ReactNode
}
