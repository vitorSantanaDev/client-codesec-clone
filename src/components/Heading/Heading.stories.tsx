import { ComponentStory, ComponentMeta } from '@storybook/react'
import Heading from '.'

export default {
  title: 'Heading',
  component: Heading,
  args: {
    children: 'Aprenda Web Hacking do zero ao avan√ßado',
    size: 'huge',
    as: 'h1'
  }
} as ComponentMeta<typeof Heading>

export const Huge: ComponentStory<typeof Heading> = (args) => (
  <Heading {...args} />
)

export const Large: ComponentStory<typeof Heading> = (args) => (
  <Heading {...args} />
)

Large.args = {
  size: 'large',
  as: 'h2'
}

export const Medium: ComponentStory<typeof Heading> = (args) => (
  <Heading {...args} />
)

Medium.args = {
  size: 'medium',
  as: 'h3'
}

export const Small: ComponentStory<typeof Heading> = (args) => (
  <Heading {...args} />
)

Small.args = {
  size: 'small',
  as: 'h4'
}
