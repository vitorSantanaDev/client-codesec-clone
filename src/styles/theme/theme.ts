export const theme = {
  colors: {
    redPrimary: '#E83752',
    redLight: '#FA5770',
    redDark: '#D4203B',
    redError: '#EB3D3D',
    redGradient: 'linear-gradient(243.46deg, #FA5770 16.66%, #E83752 83.34%)',
    redGradientDark:
      'linear-gradient(225.05deg, #FA5770 0%, rgba(232, 55, 82, 0) 100%)',
    redGradientLight:
      'linear-gradient(225.05deg, rgba(250, 87, 112, 0.1) 0%, rgba(232, 55, 82, 0) 50%)',
    greenSucsses: '#32D957',
    white: '#FFFFFF',
    whiteLight: '#F5F0F1',
    whiteDark: '#EDE4E5',
    black: '#121111',
    blackLight: '#1C1B1B',
    blackDark: '#0A0909',
    gray1: '#141414',
    gray2: '#2C2C2C',
    gray3: '#444444',
    gray4: '#5C5C5C',
    gray5: '#747474',
    gray6: '#8C8C8C',
    gray7: '#A3A3A3'
  },
  font: {
    family: {
      default: "'Raleway', sans-serif"
    },
    sizes: {
      small: '1.6rem',
      medium: '2.4rem',
      large: '3.2rem',
      xlarge: '4.0rem',
      xxlarge: '4.8rem',
      huge: '5.6rem',
      xhuge: '6.4rem',
      xxhuge: '8rem'
    }
  },
  media: {
    lteMedium: '(max-width: 768px)',
    lteLarge: '(max-width: 820px)',
    ltexLarge: '(max-width: 912px)',
    lteSmall: '(max-width: 420px)',
    lteXSmall: '(max-width: 280px)'
  },
  spacings: {
    small: '1.6rem',
    medium: '2.4rem',
    large: '3.2rem',
    xlarge: '4.0rem',
    xxlarge: '4.8rem',
    huge: '5.6rem',
    xhuge: '6.4rem',
    xxhuge: '8rem'
  }
}
