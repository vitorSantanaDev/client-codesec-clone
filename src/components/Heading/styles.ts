import styled, { css, DefaultTheme } from 'styled-components'
import { theme } from 'styles/theme/theme'

type HeadingProps = {
  as?: 'h1' | 'h2' | 'h3' | 'h4' | 'h5'
  size:
    | 'small'
    | 'medium'
    | 'large'
    | 'xlarge'
    | 'xxlarge'
    | 'huge'
    | 'xhuge'
    | 'xxhuge'
  color: string
}

const titleSize = {
  small: () => css`
    font-size: ${theme.font.sizes.small};
  `,
  medium: () => css`
    font-size: ${theme.font.sizes.medium};
  `,
  large: () => css`
    font-size: ${theme.font.sizes.large};
  `,
  xlarge: () => css`
    font-size: ${theme.font.sizes.xlarge};
  `,
  xxlarge: () => css`
    font-size: ${theme.font.sizes.xxlarge};
  `,
  huge: () => css`
    font-size: ${theme.font.sizes.huge};
  `,
  xhuge: () => css`
    font-size: ${theme.font.sizes.xhuge};
  `,
  xxhuge: () => css`
    font-size: ${theme.font.sizes.xxhuge};
  `
}

const mediaFont = (theme: DefaultTheme) => css`
  @media ${theme.media.lteMedium} {
    font-size: ${theme.font.sizes.xlarge};
  }
`

export const Wrapper = styled.h1<HeadingProps>`
  ${({ theme, size, color }) => css`
    ${titleSize[size]}
    color: ${color};
    font-weight: 600;
    font-family: ${theme.font.family.default};
    ${mediaFont(theme)}
  `}
`
