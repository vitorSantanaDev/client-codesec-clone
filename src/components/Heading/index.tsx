import { ReactNode } from 'react'
import * as S from './styles'

type HeadingProps = {
  as?: 'h1' | 'h2' | 'h3' | 'h4' | 'h5'
  size?:
    | 'small'
    | 'medium'
    | 'large'
    | 'xlarge'
    | 'xxlarge'
    | 'huge'
    | 'xhuge'
    | 'xxhuge'
  color?: string
  children: ReactNode
}

const Heading = ({
  children,
  size = 'huge',
  as = 'h1',
  color = 'white'
}: HeadingProps) => (
  <S.Wrapper as={as} color={color} size={size}>
    {children}
  </S.Wrapper>
)

export default Heading
