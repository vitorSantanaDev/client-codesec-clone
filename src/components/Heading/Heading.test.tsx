import { screen } from '@testing-library/react'
import { renderTheme } from 'styles/theme/render-theme'
import { theme } from 'styles/theme/theme'

import Heading from '.'

describe('<Heading />', () => {
  it('should render title with default styles', () => {
    const { container } = renderTheme(<Heading>Title Example</Heading>)
    const heading = screen.getByRole('heading', { name: /title example/i })
    expect(heading).toBeInTheDocument()
    expect(heading).toHaveStyle({
      color: 'white',
      'font-size': `${theme.font.sizes.huge}`
    })

    expect(container.firstChild).toMatchSnapshot()
  })
  it('should render an "h2" with the "large" font size', () => {
    const { container } = renderTheme(
      <Heading as="h2" size="large">
        Title Example
      </Heading>
    )
    const heading = document.querySelector('h2')
    expect(heading).toBeInTheDocument()
    expect(heading).toHaveStyle({
      color: 'white',
      'font-size': `${theme.font.sizes.large}`
    })

    expect(container.firstChild).toMatchSnapshot()
  })
  it('should render an "h3" with the "medium" font size', () => {
    const { container } = renderTheme(
      <Heading as="h3" size="medium">
        Title Example
      </Heading>
    )
    const heading = document.querySelector('h3')
    expect(heading).toBeInTheDocument()
    expect(heading).toHaveStyle({
      color: 'white',
      'font-size': `${theme.font.sizes.medium}`
    })

    expect(container.firstChild).toMatchSnapshot()
  })
  it('should render an "h4" with the "small" font size', () => {
    const { container } = renderTheme(
      <Heading as="h4" size="medium">
        Title Example
      </Heading>
    )
    const heading = document.querySelector('h4')
    expect(heading).toBeInTheDocument()
    expect(heading).toHaveStyle({
      color: 'white',
      'font-size': `${theme.font.sizes.medium}`
    })

    expect(container.firstChild).toMatchSnapshot()
  })
  it('should render an "h5" with the "large" font size', () => {
    const { container } = renderTheme(
      <Heading as="h5" size="large">
        Title Example
      </Heading>
    )
    const heading = document.querySelector('h5')
    expect(heading).toBeInTheDocument()
    expect(heading).toHaveStyle({
      color: 'white',
      'font-size': `${theme.font.sizes.large}`
    })

    expect(container.firstChild).toMatchSnapshot()
  })
  it('should render an "h2" with the "xlarge" font size', () => {
    const { container } = renderTheme(
      <Heading as="h2" size="xlarge">
        Title Example
      </Heading>
    )
    const heading = document.querySelector('h2')
    expect(heading).toBeInTheDocument()
    expect(heading).toHaveStyle({
      color: 'white',
      'font-size': `${theme.font.sizes.xlarge}`
    })

    expect(container.firstChild).toMatchSnapshot()
  })
  it('should render an "h2" with the "xxlarge" font size', () => {
    const { container } = renderTheme(
      <Heading as="h2" size="xxlarge">
        Title Example
      </Heading>
    )
    const heading = document.querySelector('h2')
    expect(heading).toBeInTheDocument()
    expect(heading).toHaveStyle({
      color: 'white',
      'font-size': `${theme.font.sizes.xxlarge}`
    })

    expect(container.firstChild).toMatchSnapshot()
  })
  it('should render an "h2" with the "huge" font size', () => {
    const { container } = renderTheme(
      <Heading as="h2" size="huge">
        Title Example
      </Heading>
    )
    const heading = document.querySelector('h2')
    expect(heading).toBeInTheDocument()
    expect(heading).toHaveStyle({
      color: 'white',
      'font-size': `${theme.font.sizes.huge}`
    })

    expect(container.firstChild).toMatchSnapshot()
  })
  it('should render an "h2" with the "xhuge" font size', () => {
    const { container } = renderTheme(
      <Heading as="h2" size="xhuge">
        Title Example
      </Heading>
    )
    const heading = document.querySelector('h2')
    expect(heading).toBeInTheDocument()
    expect(heading).toHaveStyle({
      color: 'white',
      'font-size': `${theme.font.sizes.xhuge}`
    })

    expect(container.firstChild).toMatchSnapshot()
  })
  it('should render an "h2" with the "xxhuge" font size', () => {
    const { container } = renderTheme(
      <Heading as="h2" size="xxhuge">
        Title Example
      </Heading>
    )
    const heading = document.querySelector('h2')
    expect(heading).toBeInTheDocument()
    expect(heading).toHaveStyle({
      color: 'white',
      'font-size': `${theme.font.sizes.xxhuge}`
    })

    expect(container.firstChild).toMatchSnapshot()
  })
  it('should render heading with "xlarge" font size when screen size is max 768px', () => {
    const { container } = renderTheme(<Heading>Title Example</Heading>)
    const heading = screen.getByRole('heading', { name: /title example/i })
    expect(heading).toBeInTheDocument()
    expect(heading).toHaveStyleRule('font-size', `${theme.font.sizes.xlarge}`, {
      media: `${theme.media.lteMedium}`
    })

    expect(container.firstChild).toMatchSnapshot()
  })
})
