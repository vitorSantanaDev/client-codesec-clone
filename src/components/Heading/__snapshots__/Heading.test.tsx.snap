// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Heading /> should render an "h2" with the "large" font size 1`] = `
.c0 {
  font-size: 3.2rem;
  color: white;
  font-weight: 600;
  font-family: 'Raleway',sans-serif;
}

@media (max-width:768px) {
  .c0 {
    font-size: 4.0rem;
  }
}

<h2
  class="c0"
  color="white"
>
  Title Example
</h2>
`;

exports[`<Heading /> should render an "h3" with the "medium" font size 1`] = `
.c0 {
  font-size: 2.4rem;
  color: white;
  font-weight: 600;
  font-family: 'Raleway',sans-serif;
}

@media (max-width:768px) {
  .c0 {
    font-size: 4.0rem;
  }
}

<h3
  class="c0"
  color="white"
>
  Title Example
</h3>
`;

exports[`<Heading /> should render an "h4" with the "small" font size 1`] = `
.c0 {
  font-size: 2.4rem;
  color: white;
  font-weight: 600;
  font-family: 'Raleway',sans-serif;
}

@media (max-width:768px) {
  .c0 {
    font-size: 4.0rem;
  }
}

<h4
  class="c0"
  color="white"
>
  Title Example
</h4>
`;

exports[`<Heading /> should render heading with "xlarge" font size when screen size is max 768px 1`] = `
.c0 {
  font-size: 5.6rem;
  color: white;
  font-weight: 600;
  font-family: 'Raleway',sans-serif;
}

@media (max-width:768px) {
  .c0 {
    font-size: 4.0rem;
  }
}

<h1
  class="c0"
  color="white"
>
  Title Example
</h1>
`;

exports[`<Heading /> should render title with default styles 1`] = `
.c0 {
  font-size: 5.6rem;
  color: white;
  font-weight: 600;
  font-family: 'Raleway',sans-serif;
}

@media (max-width:768px) {
  .c0 {
    font-size: 4.0rem;
  }
}

<h1
  class="c0"
  color="white"
>
  Title Example
</h1>
`;
